docker-compose.yml

```diff
@@ -224,6 +224,80 @@ services:
           cpus: "1.0"
           memory: 1024M
 
+  fasttaskworker1:
+    image: bluenaas-single-cell-svc
+    container_name: fasttaskworker1
+    env_file: "./.env.local"
+    build:
+      dockerfile: ./Dockerfile.dev
+    command: ["./celery.sh"]
+    environment:
+      - WORKER_NAME=fasttaskworker1
+      - PYTHONUNBUFFERED=1
+      - CELERY_BROKER_URL=${CELERY_BROKER_URL}
+      - CELERY_RESULT_BACKEND=${CELERY_RESULT_BACKEND}
+      - QUEUE_NAME=${CELERY_FAST_TASKS_QUEUE}
+    healthcheck:
+      test: "celery -A bluenaas.infrastructure.celery status || exit 1"
+      interval: 12s # The status check interval shouldn't be too low. Otherwise the worker gets busy replying to status checks & simulation performance goes down.
+      timeout: 10s # Status checks take ~7 seconds
+      retries: 5
+    volumes:
+      - ${PWD}/bluenaas:/app/bluenaas
+      # - ${PWD}/celert/start.sh:/app/bluenaas/celery.sh
+      - type: volume
+        source: nfs_share
+        target: /opt
+        volume:
+          nocopy: true
+    depends_on:
+      - redis
+      - bluenaas
+    networks:
+      - ls
+    deploy:
+      resources:
+        limits:
+          cpus: "1.0"
+          memory: 1024M
```

Comment:
If we run only one task per worker in parallel I think we should lower that. Maybe even 128 MB will be enough (compilation and the following tasks).
This will also remove some load from our dev machine(s).

Reply:
I'm noticing some strange behaviour when I lower the memory limit to 128 or 256 MB.

In these cases I notice that the worker throws the error `No such file or directory: '/opt/blue-naas/models/5/a/5acd3a33-ce68-48d5-8dcb-fb14e21872b2/x86_64/special'`. In fact, when I `ls` the `x86` directory in the volume used by the container I do not see any file called `special`, I do see it though when I run the workers with atleast 512M of memory. 

I saw that the check for the existence of this file was only added in [this PR](https://github.com/BlueBrain/Bluenaas/pull/49/files#diff-6f290d499c2affd0d40730bf399e50696209d4ab4d4a29896536e4d1a03bccddR64) by @bilalesi (this PR is not yet deployed to prod). The contents of the files `special` and `special.nrn` look good so I'll wait to get some more clarity from Bilal before removing them. These checks do not seem to affect the simulation.

Summary of results:

- 128M of memory - I see that these dedicated fast workers are very flaky. They sporadically stop responding to "pings", flower doesn't register them either sometimes.  Sometimes they do work, but throw the error for this "special" missing.

- 256M of memory - Even if I remove the check for existence of "special" file, I see "Broken pipe" error during compilation of mod files:

```
Thread Safe
Translating SKv3_1.mod into /opt/blue-naas/models/5/a/5acd3a33-ce68-48d5-8dcb-fb14e21872b2/x86_64/SKv3_1.c
Thread Safe
Translating VecStim.mod into /opt/blue-naas/models/5/a/5acd3a33-ce68-48d5-8dcb-fb14e21872b2/x86_64/VecStim.c
Thread Safe
make: *** [makemod2c_inc:30: Ih.o] Broken pipe
make: *** Waiting for unfinished jobs....
Traceback (most recent call last):
```

- 512M of memory - The models that I tested seem to build correctly.  The check for existence of "special" file in mechanisms directory also works.

Reply:
Huh, interesting. So it seems the compilation requires that much memory. Let's keep it that way then.

Thank you for the investigation, Dinika.